smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/events/MessageMarshallableGenerator.kt

```diff
@@ -50,10 +49,12 @@ class MessageMarshallableGenerator(
                             write("switch self {")
                             streamShape.eventStreamEvents(ctx.model).forEach { member ->
                                 val memberName = ctx.symbolProvider.toMemberName(member)
```

Comment:
The code below is largely just reordered to find all the bindings, then render the right Swift enum case (with or without `value` unwrapped), then actually render the bindings.

The `if-else` is really the only new code.